### PR TITLE
Handle contract redeclaration in cast script

### LINKED ISSUE #2469: Handle contract redeclaration in `sncast_std`
### Which components does the task require to be changed? (think hard pls)

sncast

### Description

We need `sncast_std` to be similar to snforges' so that if it fails with `ClassAlreadyDeclared` it carries a `ClassHash`, and we can carry on with the declaration

Or alternatively we can retrieve the state from chain and save it to the state file automatically

### PR DESCRIPTION
**Pull request by cptartur**
Closes #2469 

## Introduced changes

- Changed return type of `declare` in cast scripts so it doesn't fail on already declared contracts.

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/sncast/tests/e2e/script/declare.rs [hunk 1]
```diff
@@ -237,7 +238,7 @@ async fn test_strk_fee_settings() {
     let contract_dir = duplicate_contract_directory_with_salt(
         SCRIPTS_DIR.to_owned() + "/map_script/contracts/",
         "dummy",
-        "100",
+        "12345",
```
Comment (franciszekjob) [inline crates/sncast/tests/e2e/script/declare.rs:241]: Is this needed?

File: crates/sncast/src/main.rs [hunk 2]
```diff
@@ -256,10 +257,23 @@ async fn run_async_command(
...
+                false,
+            )
+            .await
+            .map_err(handle_starknet_command_error)
+            .map(|result| match result {
+                DeclareResponse::Success(declare_transaction_response) => {
+                    declare_transaction_response
+                }
+                DeclareResponse::AlreadyDeclared(_) => {
+                    unreachable!("Argument skip_on_already_declared is false")
```
Comment (franciszekjob) [inline crates/sncast/src/main.rs:274]: nit:
```suggestion
                    unreachable!("Argument `skip_on_already_declared` is false")
```

File: crates/sncast/tests/data/scripts/declare/src/same_contract_twice.cairo [hunk 3]
```diff
@@ -1,27 +1,48 @@
...
         "Mapa",
         FeeSettings::Eth(EthFeeSettings { max_fee: Option::Some(max_fee) }),
         Option::Some(declare_nonce)
     )
-        .unwrap_err();
+        .expect('second declare failed');
+
+    // Check if already declared contract was handled correctly
+    match second_declare_result {
+        DeclareResult::Success(_) => panic!("Should be already declare"),
```
Comment (franciszekjob) [inline crates/sncast/tests/data/scripts/declare/src/same_contract_twice.cairo:37]:
```suggestion
        DeclareResult::Success(_) => panic!("Should be already declared"),
```

**Review by franciszekjob — CHANGES_REQUESTED** [review]
Looking good 👍  left some minor remarks.

**Review by ddoktorski — APPROVED** [review]

**Review by franciszekjob — APPROVED** [review]
LGTM

**Pull request merged**